inference-engine/src/auto_plugin/auto_plugin.cpp

```diff
@@ -24,20 +26,66 @@ namespace {
         return config;
     }
 
-    DeviceInformation SelectDevice(const std::vector<DeviceInformation>& metaDevices) {
-        for (auto& item : metaDevices) {
-            if (item.deviceName.find("CPU") == 0) {
-              return item;
+    std::string GetNetworkPrecision(const InferenceEngine::CNNNetwork &network) {
+        auto nGraphFunc = network.getFunction();
+        bool isINTModel = ngraph::op::util::has_op_with_type<ngraph::op::FakeQuantize>(nGraphFunc);
+        if (isINTModel) {
+            return "INT8";
+        }
+        for (auto & node : nGraphFunc->get_ordered_ops()) {
+            auto nodeSize = node->inputs().size();
+            auto nodeName = node->get_name();
+            if (nodeSize != 0) {
+                if (nodeName.find("Convolution") != std::string::npos) {
+                    auto layerType = node->input(0).get_element_type().get_type_name();
+                    if (layerType == "f32")
+                        return "FP32";
+                    if (layerType == "f16")
+                        return "FP16";
+                }
             }
         }
```

Comment:
I think we need to support not only convolution here, but it's derivatives like here https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/transformations/src/transformations/rt_info/primitives_priority_attribute.cpp#L30-L40

Secondly, we have 2 loops actually:
1. One inside ngraph::op::util::has_op_with_type<
2. Second explicit for loop

Can we merge them together and use the second one as more generic?

Reply:
done